### PR TITLE
Set higher default for contracts maximum gas allowed

### PR DESCRIPTION
**Pull request by Stefie**
I read in one of the smart contract channels, that @kwingram25 is in the process of bigger contracts refactor. So this is just a a really basic quick fix for https://github.com/polkadot-js/apps/issues/2604 to enable people to instantiate their contracts through the UI until there's time for a more involved approach.

I just set higher default gas limit, and switched the bitlength for the gasLimit input field from u32 to u128.

### CODE REVIEW

**Comment by Stefie** [comment]
I noticed that the api is not getting the correct parameters or api-contracts itself is not handling the parameters correctly. It's using the gasLimit as the `at: BlockHash`. I was looking into it myself  but couldn't pin it down, because the error could be both in apps or the api. I can create an issue, if you could point me to the repo where the bug is most likely located.

Screenshot 2020-05-13 at 20 14 48

**Comment by jacogr** [comment]
Feel free to log, but I won't look at it for quite some time - these things are just impossible for me to play with or test and will always take the backseat to modules used on live chains where iteration is currently rapid.

But even the `Balance` parsing is broken in that RPC -

`{"id":51,"jsonrpc":"2.0","method":"contracts_call","params":[{"origin":"5CiPPseXPECbkjWCa6MnjNokrgYjMqmKndv2rSnekmSK2DjL","dest":"5CiPPseXPECbkjWCa6MnjNokrgYjMqmKndv2rSnekmSK2DjL","value":"0x0000000000000000002bdbb64bc09000","gasLimit":1000000,"inputData":"0x00"}]}`

fails with

`-32602: Invalid params: invalid type: string "0x0000000000000000002bdbb64bc09000", expected u128.`

That is properly constructed and fails. You cannot pass a `u128` over as a number in the JSON and it doesn't parse hex. so that call will never work with any non-zero balance. 

```rust
#[derive(Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
#[serde(deny_unknown_fields)]
pub struct CallRequest<AccountId, Balance> {
	origin: AccountId,
	dest: AccountId,
	value: Balance,
	gas_limit: number::NumberOrHex<u64>,
	input_data: Bytes,
}
```

TL;DR That RPC is currently unusable, despite the above error. Construction is correct based on schema.

**Comment by Stefie** [comment]
> That RPC is currently unusable, despite the above error. Construction is correct based on schema.

Is that a general problem with RPC implementation or does it only apply for the contract rpc methods?

**Comment by jacogr** [comment]
So couple of issue there -

1. Unlike the gas, it doesn't allow hex - you cannot represent numbers > 2^53-1 in JS JSON, so number generally should always allow for hex. (Which is the way it is being outputted)

2. We don't use this in other RPCs, generally everything is SCALE-encoded.

3. We had serious issues with paymentInfo to get the partial fee serialized (it was finally adjusted to only output strings). But this is out, not in.

Anyway, until fixed, nothing that can be done. The serialization is correct, the parsing thereof is not.

**Comment by Stefie** [comment]
Okay, thanks for looking into it!
